python/ray/data/_internal/logical/operators/read_operator.py

```diff
@@ -11,8 +11,23 @@ def __init__(
         self,
         datasource: Datasource,
         read_tasks: List[ReadTask],
+        estimated_num_blocks: int,
         ray_remote_args: Optional[Dict[str, Any]] = None,
     ):
-        super().__init__(f"Read{datasource.get_name()}", None, ray_remote_args)
+        if len(read_tasks) == estimated_num_blocks:
+            suffix = ""
+        else:
+            suffix = f"->SplitBlocks({int(estimated_num_blocks / len(read_tasks))})"
```

Comment:
This looks like SplitBlocks is a separate op. What about `Read(spit_blocks=N)`?

Reply:
yeah, +1 for `ReadXXX(split_blocks=N)`, otherwise `Dataset.__repr__` would become confusing.

Reply:
I'm not sure I understand this--- the original proposal is that SplitBlock is supposed to be a logical operator, since it only applies to the output of the read. It seems more clear therefore using the chaining syntax of `->` instead of making it part of the Read.

Reply:
Sorry if I miss any context, why don't we implement `SplitBlock` as a separate logical & physical operator?

The current implementation is inside `Datasource`, so it looks like part of Read & InputDataBuffer.

Reply:
I think we should, but it would get fused with Read anyways. So here we only implement it as part of Read since we have yet to decide whether it should be a general operator.

Reply:
E.g., for `dynamic_repartition()` or such.

Reply:
I see, +1 to make it a general operator.